src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c

```diff
@@ -364,8 +364,10 @@ int xocl_hot_reset(struct xocl_dev *xdev, u32 flag)
 		if (flag & XOCL_RESET_NO)
 			return 0;
 
-		xocl_peer_request(xdev, &mbreq, sizeof(struct xcl_mailbox_req),
+		mbret = xocl_peer_request(xdev, &mbreq, sizeof(struct xcl_mailbox_req),
 			&ret, &resplen, NULL, NULL, 0, 6);
+		if (mbret)
+			xocl_reset_notify(xdev->core.pdev, false);
```

Comment:
Also, we need to set any available sysfs node to indicate this error to xbutil/shim layer.
Or, use xocl_drvinst_set_offline(<pass error code>), so that xbutil reads this error code and exit from while (dev_offline) loop in xclReset() shim api.